src/librbd/internal.cc

```diff
@@ -3445,6 +3445,23 @@ int invoke_async_request(ImageCtx *ictx, const std::string& request_type,
     return r;
   }
 
+  int flush_cache(ImageCtx *ictx)
```

Comment:
I might be confused, but the ```flush(ImageCtx*)``` method does acquire all the necessary locks (as opposed to the similarly named ```_flush(ImageCtx*)``` method.  Therefore, I would just reuse the existing method.  As a bonus, it already has an associated perf counter.